### PR TITLE
Feature/cloud 1270 secrets

### PR DESCRIPTION
**Pull request by salyateem13**
# Description

Adding Okta client details to ASM and creating Okta group rule for Users group.
Fixes [#1270](https://usxtech.atlassian.net/browse/CLOUD-1270)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Tested using jazz-frontend
https://github.com/variant-inc/jazz-frontend/blob/feature/okta-app/deploy/terraform/main.tf#L35

https://octopus.apps.ops-drivevariant.com/app#/Spaces-82/projects/jazz-frontend/deployments/releases/0.1.0-okta-app0001.143/deployments/Deployments-24783?activeTab=taskLog

- [x] Sanity Testing

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added documentation to test

### CODE REVIEW

File: examples/fetching_app_secrets/app.tf [hunk 1]
```diff
@@ -0,0 +1,31 @@
+locals {
```
Comment (vibindaniel) [inline examples/fetching_app_secrets/app.tf:1]: You don't need this file for retrieving, correct?

Comment (salyateem13) [reply examples/fetching_app_secrets/app.tf:1]: Correct, I just added for demonstration purposes/testing. I can add a comment on top of the file

File: examples/fetching_app_secrets/main.tf [hunk 2]
```diff
@@ -0,0 +1,12 @@
+data "aws_secretsmanager_secret" "app_client_secret" {
```
Comment (vibindaniel) [inline examples/fetching_app_secrets/main.tf:1]: I'm very confused. Do you need all the files and variables to retrieve okta config from a secret?

File: variables.tf [hunk 3]
```diff
@@ -8,6 +8,16 @@ variable "environment" {
   type        = string
 }
 
+variable "user_tags" {
+  description = "User tags. See module docs in READ.me"
```
Comment (vibindaniel) [inline variables.tf:12]: Readme.md or Read.me?

File: outputs.tf [hunk 4]
```diff
@@ -32,3 +27,13 @@ output "app_users_group_name" {
   description = "Group name of the admin group for the Okta app."
   value       = "app-${var.environment}-${var.name}-users"
 }
+
+output "app_secrets_arn" {
+  description = "The ASM ARN of the Okta app's secrets."
```
Comment (vibindaniel) [inline outputs.tf:32]: Expand ASM please. It can also mean `American Society for Microbiology`

Comment (vibindaniel) [reply outputs.tf:32]: Call it as AWS secret manager

Comment (salyateem13) [reply outputs.tf:32]: haha okok

**Review by vibindaniel — APPROVED** [review]

File: variables.tf [hunk 5]
```diff
@@ -242,8 +247,14 @@ variable "skip_groups" {
...
-  type        = list(any)
+variable "trusted_origins" {
+  description = "Urls to add as trusted origins to Okta Security"
+  type        = list(string)
   default     = []
 }
+
+variable "app_user_group_rule_group" {
+  description = "Okta group name. Users of this group will be added to your Okta app's user group"
+  type        = string
```
Comment (vibindaniel) [inline variables.tf:258]: Can this be list of groups?

Comment (salyateem13) [reply variables.tf:258]: Sure

Comment (salyateem13) [reply variables.tf]: Supporting multiple group assignments

https://github.com/variant-inc/jazz-frontend/blob/106c38ecdedff6ed611d1ea4f1fac72aed2fcf04/deploy/terraform/main.tf#L35

https://octopus.apps.ops-drivevariant.com/app#/Spaces-82/projects/jazz-frontend/deployments/rel[…]43/deployments/Deployments-24783?activeTab=taskLog

https://office-drivevariant-admin.oktapreview.com/admin/groups#rules

**Review by vibindaniel — APPROVED** [review]

**Pull request merged**